packages/stores/src/queries/fallback-query-store.ts

```diff
@@ -0,0 +1,34 @@
+import { observable, autorun, makeObservable, runInAction } from "mobx";
+import { ObservableQuery } from "@keplr-wallet/stores";
+
+/** Wraps an ordered list of similar query stores and falls back to the first store that returns a valid response. */
+export class FallbackStore<TStore extends ObservableQuery> {
+  @observable.ref
+  protected _responsiveStore: TStore;
+
+  constructor(readonly orderedStores: TStore[]) {
+    this._responsiveStore = orderedStores[0];
+
+    // Runs any time error changes in current store.
+    autorun(() => {
+      if (this._responsiveStore.error) {
+        // If the current store has an error, fallback to the next store.
+        const index = orderedStores.indexOf(this._responsiveStore);
+        if (index < orderedStores.length - 1) {
+          runInAction(() => (this._responsiveStore = orderedStores[index + 1]));
+        } else {
+          console.warn(
+            "FallbackStore ran out of fallback stores. Last store error:",
+            this._responsiveStore.error
+          );
```

Comment:
Maybe we should restart the loop by setting the responsive store to index 0, so we don't end up with a dead end.

Reply:
Wouldn't that cause an infinite loop? Or do we add a flag to remember that we arrived at the end, and are defaulting to the first?

Reply:
I assume the query will keep retrying between all the stores until one is successful. Do you think that would crash the app?

Reply:
Eh I think it would possibly make the page unresponsive. I think we can't assume the prev failed services will come back online as the user is sitting with the window open. We should assume the last fallback store won't fail, that should be the edge case.